lib/avro-patches/schema_validator/schema_validator.rb

```diff
@@ -64,15 +64,55 @@ def to_s
 
     class << self
       def validate!(expected_schema, datum)
-        result = Result.new
-        validate_recursive(expected_schema, datum, ROOT_IDENTIFIER, result)
-        fail ValidationError, result if result.failure?
-        result
+        with_result do |result|
+          validate_recursive(expected_schema, datum, ROOT_IDENTIFIER, result)
+        end
+      end
+
+      def validate_simple!(expected_schema, datum)
```

Comment:
It does need to be part of the public API for this class so that it can be called from `Avro::Schema.validate_simple`.